### ISSUE TITLE
Chakra default input placeholder isn't WCAG compliant

### ISSUE THREAD

**Issue by andy-zhou** [chakra-ui/chakra-ui#5886]
### Description

The default input placeholder is not accessible

### Link to Reproduction

n/a

### Chakra UI Version

1.8.8

### Operating System

- [X] macOS
- [X] Windows
- [X] Linux

### Additional Information

The input example on the website has a color of #A3ADBF, on a background of #FFFFFF. This has a contrast ratio of 2.26:1, which fails WCAG accessibility standards.

While this is easily worked around by adding a label, I noticed that even usages on the website itself (such as the search bar), don't follow relevant a11y guidelines.

I don't have a great solution to this, since I just realized this myself and high-contrast placeholders look admittedly ugly, but I did find some comparisons across the web:

Examples of WCAG-compliant placeholders:
- Figma login form

    <img width="343" alt="image" src="https://user-images.githubusercontent.com/3608830/163685532-7ceb151c-51d0-492f-b580-6e9f9f465428.png">

- Airbnb login form

    <img width="544" alt="image" src="https://user-images.githubusercontent.com/3608830/163685554-a0f9c191-42f0-48f1-969e-2e4cbdef081a.png">

Examples of WCAG-failing placeholders:
- Reddit login form (arguable because of bolded text, unsure about a11y requirements for small bolded text)

    <img width="307" alt="image" src="https://user-images.githubusercontent.com/3608830/163685579-1eff7956-1ee1-4dbd-8c2e-4f70d3bef949.png">

**Comment by TylerAPfledderer** [comment]
This is a suggested solution from W3: https://www.w3.org/WAI/tutorials/forms/instructions/#styling

As to the searchbar in the docs, I already had a PR in mind as there is a simple prop adjustment to make the text and icon colors compliant! 😁

**Comment by TylerAPfledderer** [comment]
As to the placeholder styling rendered for the package, this is a part of the [global theme](https://github.com/chakra-ui/chakra-ui/blob/next/packages/theme/src/styles.ts#L13-L15). The above W3 suggestion could also be used.

Something like this:
```js
"*::placeholder": {
  color: mode("gray.600", "gray.400")(props), // If you want to stick with the tokens!
  opacity: 1
},
```
